src/core/qgsstringstatisticalsummary.cpp

```diff
@@ -75,6 +77,21 @@ void QgsStringStatisticalSummary::addValue( const QVariant &value )
 void QgsStringStatisticalSummary::finalize()
 {
   mMeanLength = mSumLengths / static_cast< double >( mCount );
+
+  if ( mStatistics & Minority || mStatistics & Majority )
+  {
+    QList<int> valueCounts = mValues.values();
+    std::sort( valueCounts.begin(), valueCounts.end() );
```

Comment:
This is quite an overhead with a complexity of `n log(n)` instead of `n`.

```
double min = *std::min_element(valueCounts.begin(), valueCounts.end());
double max = *std::max_element(valueCounts.begin(), valueCounts.end());
```

Reply:
It was shameless copy-paste from here https://github.com/qgis/QGIS/blob/master/src/core/qgsstatisticalsummary.cpp#L231-L240, I guess this optimization is applicable there as well?

Reply:
Pretty much. If you can do that too, that's appreciated :+1:

Reply:
Done! Thanks a lot for the feedback! Every time I learn a new bit.